Feature/8092 add missing profiler steps

JorgeMucientes opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #8092
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
This PR is a continuation of[ this other](https://github.com/woocommerce/woocommerce-android/pull/8009). It adds the missing profiler steps:
- Commerce Journey Step
- Ecommerce platforms step

**Couple things to keep in mind:** 

- Profiler steps are still under feature flag and only available for debug builds as we are still discussing on the options that should be displayed. This PR is also missing localization for the displayed options. Once we have the definitive list of options to be displayed we can proceed to add them as strings to get them localized. More context here p1672316116056879-slack-C045CUK1Y3U

- This PR doesn't include saving the profiler options in backend because the API is still WIP.

- The scope of these changes don't include adding the Country selection screen. So right now after the profiler steps are done, user will land on domain picker screen.

### Testing instructions
- Trigger store creation flow
- Navigate through profiler steps selecting different option and verifying the navigation between complies with the expected design: 

![Screenshot 2022-12-30 at 20 23 07](https://user-images.githubusercontent.com/2663464/210105445-c5478f8a-6281-4237-9695-90a03154f484.png)

- Check that selected state is correctly displayed in dark mode, following this design:

![Screenshot 2022-12-30 at 20 25 12](https://user-images.githubusercontent.com/2663464/210105593-af23ef38-e4f4-4536-b9e1-f9edd0cad1ce.png)

I will be AFF during the next week. So, if there's anything blocking I won't be able to fix it until the 9th January. Feel free to apply any changes you consider and resume work from here if needed @0nko

JorgeMucientes committed:
Make existing store profiler code more generic

Profiler steps all share the same UI structure, so it makes sense to make the existing code more generic in order to re use it for different profiler options

JorgeMucientes committed:
Fix wrong background for selected item in dark mode

JorgeMucientes committed:
Load profiler data from json string

JorgeMucientes committed:
Remove unused code

JorgeMucientes committed:
Update outdated text copies

JorgeMucientes committed:
Add copies for new steps

JorgeMucientes committed:
Handle navigation to next profiler step after selecting site category

JorgeMucientes committed:
Update new store data with selected category

JorgeMucientes committed:
Save selected profiler options to new store model

JorgeMucientes committed:
Refactor into independent fragments and view models

Attempted to use a single view model and fragment for all 3 profiler steps. But this ended in several navigation issue I could not handle in a clean way. Also the code was starting to get complex. Refactored into much simpler independent view model and fragments.

JorgeMucientes committed:
Adds tracking to profiler steps viewed

0nko commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/mystoresummary/MyStoreSummaryViewModel.kt:
```diff
@@ -16,14 +16,14 @@ import javax.inject.Inject
             name = newStore.data.name,
             domain = newStore.data.domain ?: "",
-            category = newStore.data.category,
+            category = newStore.data.industry,
```
NP: the parameter should be updated to `industry` to avoid confusion.

0nko commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/profiler/StoreProfilerEcommercePlatformsViewModel.kt:
```diff
@@ -0,0 +1,62 @@
+    private val storeProfilerRepository: StoreProfilerRepository,
+    private val resourceProvider: ResourceProvider,
+) : BaseStoreProfilerViewModel(savedStateHandle, newStore) {
+    override val profilerStep: ProfilerOptionType = COMMERCE_JOURNEY
```
This should be `ECOMMERCE_PLATFORM`

JorgeMucientes replied:
Good catch! Although I ended up removing the `profilerStep` field completely.

0nko commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/profiler/StoreProfilerEcommercePlatformsViewModel.kt:
```diff
@@ -0,0 +1,62 @@
+
+    private fun Platform.toStoreProfilerOptionUi() =
+        StoreProfilerOptionUi(
+            type = ECOMMERCE_PLATFORM,
```
I like how you used the `profilerStep` property in the othe VMs, which lowers the risk of using the wrong value here.

JorgeMucientes replied:
I actually ended up removing this `profilerStep` field from everywhere is [it wasn't needed. ](https://github.com/woocommerce/woocommerce-android/pull/8097/commits/6554c70476a4ae5b6ba0a82159f2127810ff7059)
I think that field was more of a leftover from a previous implementation where I tried having everything in a single view model.

0nko commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/NewStore.kt:
```diff
@@ -11,7 +11,9 @@ class NewStore @Inject constructor() {
         domain: String? = null,
-        category: String? = null,
+        industry: String? = null,
+        commercJourney: String? = null,
```
NP: Typo.

0nko reviewed (approved):
Everything is looking good, great job! I found a couple of minor issues but once those are fixed and conflicts resolved, it's good to go! 👍

JorgeMucientes committed:
Fix wrong profiler step initialisation

JorgeMucientes committed:
Remove unneeded profiler step param from title and description methods

JorgeMucientes committed:
Remove profilerOptionType completely

JorgeMucientes committed:
Merge branch 'trunk' into feature/8092-add-missing-profiler-steps

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/profiler/StoreProfilerEcommercePlatformsFragment.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/profiler/StoreProfilerScreen.kt
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/storecreation/profiler/StoreProfilerViewModel.kt

JorgeMucientes committed:
Remove empty StoreProfilerViewModel.kt after merge with trunk

JorgeMucientes committed:
Fix help button navigation fro profiler steps

JorgeMucientes committed:
Fix typo

JorgeMucientes committed:
Rename field for consistency with profiler steps code

JorgeMucientes committed:
Fix detekt indentation issues

JorgeMucientes commented:
Thanks for the review @0nko 🙏🏼 

I've addressed all the feedback, plus added a couple minor refactors to remove unneeded code. Ready for another round.

JorgeMucientes committed:
Merge branch 'trunk' into feature/8092-add-missing-profiler-steps

# Conflicts:
#	WooCommerce/src/main/res/values/strings.xml

JorgeMucientes commented:
As this needs to be merged for milestone `11.9` and the latest changes only add suggestions from @0nko (that is AFK) I'll proceed with merging.
